docs/src/main/paradox/json.md

```diff
@@ -0,0 +1,40 @@
+JSON marshalling
+----------------
+
+When exercising the app, you interacted with JSON payloads. How does the example app convert data between JSON format and data that can be used by Scala classes? The answer begins in the server class definition `JsonSupport` trait:
+
+@@snip [UserRoutes.scala]($g8src$/scala/com/lightbend/akka/http/sample/UserRoutes.scala) { #user-routes-class }
+
+This trait is implemented in the `JsonSupport.scala` source file:
+
+@@snip [JsonSupport.scala]($g8src$/scala/com/lightbend/akka/http/sample/JsonSupport.scala)
+
+We're using the [Spray JSON](https://github.com/spray/spray-json) library here, which allows us to define json marshallers
+(or `formats` how Spray JSON calls them) in a type-safe way. In other words, if we don't provide a format instance for 
+a type, yet we'd try to return it in a route by calling `complete(someValue)` the code would not compile - saying that
+it does not know how to marshal the `SomeValue` type. This has the up-side of us being completely in control over what 
+we want to expose, and not exposing some type accidentally in our API.
+
+To handle the two different payloads, the trait defines two implicit values; `userJsonFormat` and `usersJsonFormat`. Defining the formatters as `implicit` ensures that the compiler can map the formatting functionality with the case classes to convert.
+
+The `jsonFormatX` methods come from Spray JSON. The `X` represents the number of parameters in the underlying case classes:
+
+@@snip [UserRegistryActor.scala]($g8src$/scala/com/lightbend/akka/http/sample/UserRegistryActor.scala) { #user-case-classes }
+
+We won't go into how the formatters are implemented - this is done for us by the library. All you need to remember for now is to define the formatters as implicit and that the formatter used should map the number of parameters belonging to the case class it converts.
+
+@@@ note
```

Comment:
What? It worked for me and is part of paradox: http://developer.lightbend.com/docs/paradox/latest/features/callouts.html#note-callout

Will double check